cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -857,4 +873,41 @@ parquet_metadata read_parquet_metadata(host_span<std::unique_ptr<datasource> con
                           metadata.get_rowgroup_metadata()};
 }
 
+void reader::impl::update_output_nullmasks_for_pruned_pages(cudf::host_span<bool const> page_mask)
+{
+  auto const& subpass    = _pass_itm_data->subpass;
+  auto const& pages      = subpass->pages;
+  auto const& chunks     = _pass_itm_data->chunks;
+  auto const num_columns = _input_columns.size();
+
+  CUDF_EXPECTS(pages.size() == page_mask.size(), "Page mask size mismatch");
+
+  thrust::for_each(
+    thrust::make_zip_iterator(thrust::make_tuple(pages.host_begin(), page_mask.begin())),
+    thrust::make_zip_iterator(thrust::make_tuple(pages.host_end(), page_mask.end())),
```

Comment:
I think this is usually done like this:
```suggestion
auto page_mask_begin = thrust::make_zip_iterator(thrust::make_tuple(pages.host_begin(), page_mask.begin());

  thrust::for_each(
    page_mask_begin,
    page_mask_begin + pages.size(),
```

Reply:
Moving this suggestion to #18011